keras_nlp/src/models/stable_diffusion_3/stable_diffusion_3_backbone.py

```diff
@@ -0,0 +1,191 @@
+        )
+        pooled_output = ops.take_along_axis(inputs, indices[:, :, None], axis=1)
+        pooled_output = ops.squeeze(pooled_output, axis=1)
+        projection_output = self.dense(pooled_output)
+        return projection_output
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "hidden_dim": self.hidden_dim,
+            }
+        )
+        return config
+
+
+@keras_nlp_export("keras_nlp.models.StableDiffusion3Backbone")
+class StableDiffusion3Backbone(Backbone):
+    def __init__(
+        self,
+        mmdit_patch_size,
+        mmdit_num_heads,
+        mmdit_hidden_dim,
+        mmdit_depth,
+        mmdit_position_size,
+        vae_stackwise_num_filters,
+        vae_stackwise_num_blocks,
+        clip_l,
+        clip_g,
+        t5=None,
+        latent_channels=16,
+        output_channels=3,
+        num_train_timesteps=1000,
+        shift=1.0,
+        height=None,
+        width=None,
+        data_format=None,
+        dtype=None,
+        **kwargs,
+    ):
+        height = int(height or 1024)
+        width = int(width or 1024)
+        if height % 8 != 0 or width % 8 != 0:
+            raise ValueError(
+                "`height` and `width` must be divisible by 8. "
+                f"Received: height={height}, width={width}"
+            )
+        data_format = standardize_data_format(data_format)
+        if data_format != "channels_last":
+            raise NotImplementedError
+        latent_shape = (height // 8, width // 8, latent_channels)
+
+        # === Layers ===
+        self.clip_l = clip_l
+        self.clip_l_projection = Projection(
+            clip_l.hidden_dim, dtype=dtype, name="clip_l_projection"
+        )
+        self.clip_l_projection.build([None, clip_l.hidden_dim], None)
+        self.clip_g = clip_g
+        self.clip_g_projection = Projection(
+            clip_g.hidden_dim, dtype=dtype, name="clip_g_projection"
+        )
+        self.clip_g_projection.build([None, clip_g.hidden_dim], None)
+        self.t5 = t5
+        self.mmdit = MMDiT(
+            mmdit_patch_size,
+            mmdit_num_heads,
+            mmdit_hidden_dim,
+            mmdit_depth,
+            mmdit_position_size,
+            latent_shape=latent_shape,
+            data_format=data_format,
+            dtype=dtype,
+            name="mmdit",
+        )
+        self.vae = VAEImageDecoder(
+            vae_stackwise_num_filters,
+            vae_stackwise_num_blocks,
+            output_channels,
+            latent_shape=latent_shape,
+            data_format=data_format,
+            dtype=dtype,
+            name="vae",
+        )
+        self.noise_scheduler = FlowMatchEulerDiscreteScheduler(
+            num_train_timesteps=num_train_timesteps,
+            shift=shift,
+        )
+
+        # === Functional Model ===
+        # TODO: Can we define the model here?
+        super().__init__(dtype=dtype, **kwargs)
```

Comment:
We would like to keep this a function model. What part doesn't work if we try?

Reply:
I believe this function model should be incorporated into `fit`. However, I'm currently unsure how to train `StableDiffusion3`.
It could be a future work. WDYT?